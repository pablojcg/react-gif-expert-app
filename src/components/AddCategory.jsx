
import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('');

    // const onInputChange = ({ target }) => {
    //     setInputValue(target.value);
    // };

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onAddCategory(inputValue.trim());
        setInputValue('');
    };

    return (
        <form onSubmit={ onSubmit } aria-label='form'>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={(e) => setInputValue(e.target.value)}
            />
        </form>
    );
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired
};