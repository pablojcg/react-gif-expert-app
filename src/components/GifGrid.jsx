import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { gifs, isLoading } = useFetchGifs(category);

    return (
        <>  
            <h3>{category}</h3>
            
            { !isLoading ?
                <>
                    <div className="card-grid">
                        {gifs.map((image) => (
                            <GifItem
                                key={image.id}
                                {...image}
                            />
                        ))}
                    </div>
                </>
                : <h2>Cargando...</h2>}
        </>
    );
} 

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
