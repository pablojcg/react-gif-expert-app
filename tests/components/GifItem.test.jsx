import { GifItem } from "../../src/components/GifItem";
import { render, screen } from "@testing-library/react";

describe('Probando Componente <GifItem />', () => {

    const title = 'Un título';
    const url = 'https://localhost/algo.jpg';
    const id = '123';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} id={id} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        const { getByRole } = render(<GifItem title={title} url={url} id={id} />);
        const { src, alt } = getByRole('img');
        const { src: srcS, alt: altS } = screen.getByRole('img');
        //screen.debug();
        expect(srcS).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el título en el componente', () => {
        render(<GifItem title={title} url={url} id={id} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});  