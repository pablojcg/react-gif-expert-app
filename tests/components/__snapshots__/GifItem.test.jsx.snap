// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Probando Componente <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un título"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un título
    </p>
  </div>
</div>
`;
